code/game/objects/items/gift.dm

```diff
@@ -1,116 +1,123 @@
+		user.investigate_log("has unwrapped a present containing [thing.type].", INVESTIGATE_PRESENTS)
+		user.put_in_hands(thing)
+		thing.add_fingerprint(user)
 
-	var/gift_type = pick(gift_type_list)
+	qdel(src)
 
-	return gift_type
+/obj/item/gift/proc/get_gift_type()
+	var/static/list/gift_type_list = null
+
+	if(isnull(gift_type_list))
+		gift_type_list = list(
+			/obj/item/banhammer,
+			/obj/item/bikehorn,
+			/obj/item/book/manual/chef_recipes,
+			/obj/item/book/manual/wiki/barman_recipes,
+			/obj/item/clothing/head/costume/snowman,
+			/obj/item/clothing/neck/tie/horrible,
+			/obj/item/clothing/suit/costume/poncho,
+			/obj/item/clothing/suit/costume/poncho/green,
+			/obj/item/clothing/suit/costume/poncho/red,
+			/obj/item/clothing/suit/costume/snowman,
+			/obj/item/clothing/suit/jacket/leather,
+			/obj/item/clothing/suit/jacket/leather/biker,
+			/obj/item/food/grown/ambrosia/deus,
+			/obj/item/food/grown/ambrosia/vulgaris,
+			/obj/item/grenade/smokebomb,
+			/obj/item/grown/corncob,
+			/obj/item/instrument/guitar,
+			/obj/item/instrument/violin,
+			/obj/item/lipstick/random,
+			/obj/item/pai_card,
+			/obj/item/pen/invisible,
+			/obj/item/pickaxe/diamond,
+			/obj/item/poster/random_contraband,
+			/obj/item/poster/random_official,
+			/obj/item/soap/deluxe,
+			/obj/item/sord,
+			/obj/item/stack/sheet/mineral/coal,
+			/obj/item/storage/backpack/holding,
+			/obj/item/storage/belt/champion,
+			/obj/item/storage/belt/utility/full,
+			/obj/item/storage/box/snappops,
+			/obj/item/storage/crayons,
+			/obj/item/storage/photo_album,
+			/obj/item/storage/wallet,
+			/obj/item/toy/basketball,
+			/obj/item/toy/beach_ball,
+		)
+
+		gift_type_list += subtypesof(/obj/item/clothing/head/collectable)
+		//Add all toys, except for abstract types and syndicate cards.
+		gift_type_list += subtypesof(/obj/item/toy) - (((typesof(/obj/item/toy/cards) - /obj/item/toy/cards/deck) + /obj/item/toy/figure + /obj/item/toy/ammo))
 
+	var/gift_type = pick(gift_type_list)
+	return gift_type
 
-/obj/item/a_gift/anything
+/// Gifts that typically only very OP stuff or admins or Santa Claus himself should be giving out, as they contain ANY valid subtype of `/obj/item`, including stuff like instagib rifles. Wow!
+/obj/item/gift/anything
 	name = "christmas gift"
 	desc = "It could be anything!"
 
-/obj/item/a_gift/anything/get_gift_type()
-	if(!GLOB.possible_gifts.len)
-		var/list/gift_types_list = subtypesof(/obj/item)
-		for(var/V in gift_types_list)
-			var/obj/item/I = V
-			if((!initial(I.icon_state)) || (!initial(I.inhand_icon_state)) || (initial(I.item_flags) & ABSTRACT))
-				gift_types_list -= V
-		GLOB.possible_gifts = gift_types_list
-	var/gift_type = pick(GLOB.possible_gifts)
+/obj/item/gift/anything/get_gift_type()
+	var/static/list/obj/item/possible_gifts = null
+
+	if(isnull(possible_gifts))
+		possible_gifts = list()
+		for(var/type in subtypesof(/obj/item))
+			var/obj/item/thing = type
```

Comment:
```suggestion
		for(var/obj/item/thing as anything in subtypesof(/obj/item))
```

Reply:
This doesnt work, had to revert in latest commit